### ISSUE TITLE
ChatSecure asking me for password to unlock after restore

### ISSUE THREAD

**Issue by n00b12345** [ChatSecure/ChatSecure-iOS#764]
I just restored an iPad and I clearly remember not setting an unlock code for ChatSecure.

However, after the restore process, it's asking me for an unlock code. I'm reinstalling ChatSecure and it's not a big deal, but I just wanted to know if this is by design.

Thanks.

**Comment by chrisballinger** [comment]
For your safety the encryption key used for the local database is set to
only backup via encrypted iTunes backups to the same device. Otherwise a
compromised backup could be used to extract your conversation history, etc.
Sorry for the inconvenience!

**Comment by n00b12345** [comment]
@chrisballinger that is ingenuous. But also something I should keep in mind when I upgrade my iPhones and iPads in the future. 

Thanks for the help. The security features are most certainly impeccable.

**Comment by dholl** [comment]
A friend still has their old phone in possession, but they bought a new phone and Apple auto-installed ChatSecure to their new phone.
When he tried opening ChatSecure, it asked for his passphrase, which he thought was his password on the Prosody server.  After we reset his password, then I realized that "passphrase" might be referring to some internal encrypted storage on ChatSecure.  So far a quick web search, I ended up here.

Sure enough, removing and reinstalling ChatSecure was enough to get him running again, understandably without his chat history.

But just in case this were to happen again with a different person who also still has their old phone and are transitioning to a new phone (or maybe my friend buys yet another new phone a couple years from now), is there an area within ChatSecure to look up the current passphrase on the old phone, so that a user may enter it into the new phone?  (possibly even getting fancy via a QR code to transfer the passphrase quickly?)

Thank you very much for this project!

- David
